Support for Typed POJO

I have an object that is generically typed, for example:

```kotlin
import com.github.pozo.KotlinBuilder

interface Bar

@KotlinBuilder
data class Foo<T : Bar>(
   val bar: T
)
```

When I build this, it generates the following:

```java
public final class FooBuilder {
  private T bar;

  public FooBuilder setBar(T bar) {
    this.bar = bar;
    return this;
  }

  public Foo create() {
    return new Foo(bar);
  }

  public static FooBuilder builder() {
    return new FooBuilder();
  }
}
```

But it also dumps the following error:
```
  private T bar;
          ^
  symbol:   class T
  location: class FooBuilder/.../FooBuilder.java:6: error: cannot find symbol
  public FooBuilder setBar(T bar) {
                           ^
  symbol:   class T
  location: class FooBuilder
```

This seems very close and is simply missing the same type restriction on the Builder class.

--- a/mapstruct-kotlin-processor/src/main/kotlin/com/github/pozo/BuilderProcessor.kt
+++ b/mapstruct-kotlin-processor/src/main/kotlin/com/github/pozo/BuilderProcessor.kt
@@ -8,6 +8,7 @@ import com.github.pozo.BuilderGenerator.readHeader
 import com.google.auto.service.AutoService
 import com.squareup.javapoet.JavaFile
 import com.squareup.javapoet.TypeSpec
+import com.squareup.javapoet.TypeVariableName
 import kotlinx.metadata.Flag
 import kotlinx.metadata.Flags
 import kotlinx.metadata.KmClassVisitor
@@ -66,6 +67,7 @@ class BuilderProcessor : AbstractProcessor() {
 
         with(classBuilder) {
             this.addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+            this.addTypeVariables(typeElement.typeParameters.map { TypeVariableName.get(it) })
             val constructors = ElementFilter.constructorsIn(typeElement.enclosedElements)
 
             constructors.stream()

